standalone/pherry/src/main.rs

```diff
@@ -683,6 +683,18 @@ async fn register_worker(
 
 const DEV_KEY: &str = "0000000000000000000000000000000000000000000000000000000000000001";
 
+async fn wait_until_synced(client: &XtClient) -> Result<()> {
+    loop {
+        let state = client.rpc.system_sync_state().await?;
+        if let Some(highest) = state.highest_block {
+            if highest - state.current_block <= 1 {
+                return Ok(());
+            }
+        }
+        delay_for(Duration::from_secs(10)).await;
```

Comment:
I think we either wait shorter (e.g. 1s) here, or we allow a slightly higher offset between `highest` and `current_block` (e.g. 2 blocks), to avoid occasional long waiting in some edge case.

Reply:
Fixed